### ISSUE TITLE
Blog Date Accessibility

### ISSUE THREAD

**Issue by rutterjt** [rutterjt/v1#17]
The dates on blog posts and blog post excerpts should be wrapped in a `<time>` tag.

This will involve modifying the graphql queries, since the date values currently being returned are formatted as human readable strings (e.g., 'June 17, 2022').

1. Modify graphql queries to return a date string (e.g., '2022-06-17')
2. Make this date string the value for `<time datetime>`
3. Use a library to format the date string as a human readable string

**Comment by rutterjt** [comment]
Running into a snag. Gatsby apparently returns all dates in the UTC time zone (e.g., if the post date is 2022-06-15, graphql returns a date string of '2022-06-15T00:00:00.000Z'). This means that in my timezone, all post dates appear to be one day early.

I also tried storing the dates in my frontmatter as strings, ('2022-06-15' instead of 2022-06-15), which avoided Gatsby returning them as UTC, but unfortunately the `Date()` constructor treats all [date-only strings as UTC](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#parameters). So it's the same problem for a different reason.

**Comment by rutterjt** [comment]
Found a solution using `date-fns`. Decided to store the dates in blog frontmatter as partial ISO strings ('2022-06-15'), then parse them with `date-fns`:

```
import parseISO from 'date-fns/parseISO';
import format from 'date-fns/format';

// given node.frontmatter.date === '2022-06-15'
<time dateTime={node.frontmatter.date}>
  {format(parseISO(node.frontmatter.date), 'MMMM d, yyyy')}
</time>
```

This works because unlike the `Date` constructor, `parseISO` always returns dates in the local timezone.
